nix/package.nix

```diff
@@ -38,19 +42,44 @@ writeShellScriptBin "agenix" ''
     exit 1
   }
 
-  case "$1" in
-    "help"|"--help"|"-help"|"-h")
-      show_help
-      exit 1
-      ;;
-
-    ${lib.concatStringsSep "|" allApps})
-      APP=$1
-      shift
-      echo "Collecting information about hosts. This may take a while..."
-      exec nix run .#agenix-rekey.apps.${lib.escapeShellArg stdenv.hostPlatform.system}."$APP" -- "$@"
-      ;;
-
-    *) die "Unknown command: $1" ;;
-  esac
+  APP=""
+  SHOW_TRACE_ARG=""
+  # Various Bash versions treat empty arrays as unset, which then trigger
+  # unbound variable errors.
+  PASS_THRU_ARGS=()
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      "help"|"--help"|"-help"|"-h")
+        show_help
+        exit 1
+        ;;
+      "--show-trace")
+        # It is potentially desirable to use --show-trace in the subcommand as
+        # well as this command.  To do so, the --show-trace argument must be
+        # provided before (agenix) or after (subcommand) to indicate which one
+        # is to be used.  We account for this here.
+        if [[ "$APP" == "" ]]; then
+          SHOW_TRACE_ARG='--show-trace'
```

Comment:
As part of some testing today I noticed that `--help` doesn't propagate to the subcommand as one would think (ex: `agenix rekey --help`) and instead just shows the `agenix` help.  I think `--help` will need similar treatment we're giving `--show-trace` here.

I'd like to address that in a separate PR if that's all the same to you.  When I do so, I am thinking I will change this `SHOW_TRACE_ARG` to be a more generalized `COMMAND_PASS_THRU_ARGS` array and rename `PASS_THRU_ARGS` to `SUBCOMMAND_PASS_THRU_ARGS`.  Thoughts?

Reply:
Sounds good!